### PR TITLE
Group Labs by Location

### LINKED ISSUE #1116: Group labs by location
## Action Requested
Group lab results by the lab they were performed at, and add information about the lab location.

>eCR Viewer
>> Lab Info
>>> Lab Results
>>>> Lab Performing Name
>>>> Lab Address
>>>> Lab Contact
>>>>> (individual lab results)

## Acceptance Criteria
All lab observations in an eCR are grouped by the lab they were performed at. 

Each location has performing name, address, and contact fields.

## Testing
Unit testing will be required.

## Background & Context
Grouping lab results by the location performed will hopefully make the labs easier to scan through at a glance.

## Designs
[Figma Designs](https://www.figma.com/file/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?type=design&node-id=1024%3A136062&mode=design&t=lClwL7CnvrviunUF-1)
[Figma Clickable Prototype](https://www.figma.com/proto/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?page-id=871%3A2283&type=design&node-id=1606-27619&viewport=1011%2C558%2C0.05&t=sGUJKIN0MXVFYIfm-1&scaling=min-zoom&starting-point-node-id=1606%3A27619&mode=design)

## Design Review
Things to look for:
- there should be a new section for every different lab facility that tests came in from, this should also be reflected in the left sidebar navigation
- the main header for each of these lab groupings should be: "Lab Results from [insert lab name here]"
- ensure field names, ordering of fields, and capitalization of field names matches the design
- make sure there is a gray diving line below each field
- ensure spacing looking visually similar to designs

## Additional Information
We _should_ already be converting all of the above information to FHIR (they should be within the Observation resources), but grouping by location may take additional work.

### PR DESCRIPTION
**Pull request by KennethSkylight**
# PULL REQUEST

## Summary
Labs have a performer associated with them. We are now grouping them by organization and displaying the data to the user. The logic gets a bit complex so any suggestions on how to make it more readable will be good.

## Related Issue
Fixes #1116 

## Screenshots

The sample data is missing a phone number
image

## How to test
Use file 0a0b1da9-7ac6-4668-a448-c66012afae3b to test so 
http://localhost:3000/view-data?id=0a0b1da9-7ac6-4668-a448-c66012afae3b

### CODE REVIEW

File: containers/ecr-viewer/src/app/view-data/components/LabInfo.tsx [hunk 1]
```diff
@@ -19,40 +19,44 @@ interface LabInfoProps {
...
-            <div className={"margin-left-auto"}>
-              <ExpandCollapseButtons
-                id={"lab-info"}
-                buttonSelector={"h5 > .usa-accordion__button"}
-                accordionSelector={".accordion-rr > .usa-accordion__content"}
-                expandButtonText={"Expand all labs"}
-                collapseButtonText={"Collapse all labs"}
-              />
+        {labResults.map((labResult, labResultIndex) => (
+          <div key={labResultIndex}>
```
Comment (BobanL) [inline containers/ecr-viewer/src/app/view-data/components/LabInfo.tsx:27]: I think [best practices for keys is not to use the index](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key:~:text=You%20might%20be,and%20confusing%20bugs.) if we don't have to. Could we just use `labResult.organizationId` instead?

File: containers/ecr-viewer/src/app/view-data/components/tests/LabInfo.test.tsx [hunk 2]
```diff
@@ -3,34 +3,57 @@ import LabInfo from "@/app/view-data/components/LabInfo";
...
+            />,
+          ],
+          organizationDisplayData: [labinfoOrg[1]],
+          organizationId: "efgh",
+        },
       ]}
     />
   );
   it("should hide all labs when collapse button is clicked", async () => {
-    const user = userEvent.setup();
```
Comment (BobanL) [inline containers/ecr-viewer/src/app/view-data/components/tests/LabInfo.test.tsx:28]: Was there a reason why this userEvent setup was removed? I think its the [recommend approach to use what is returned from `userEvent.setup()`](https://testing-library.com/docs/user-event/setup/#:~:text=Using%20the%20methods%20on%20the%20instances%20returned%20by%20userEvent.setup()%20is%20recommended.)

**Review by BobanL — COMMENTED** [review]
A couple of quick nits - but otherwise looks good 👍

File: containers/ecr-viewer/src/app/api/fhirPath.yml [hunk 3]
```diff
@@ -104,3 +104,13 @@ observationComponent: "code.coding.display"
...
+
+# Organization
+organizations: "Bundle.entry.resource.where(resourceType = 'Organization')"
+organizationId: "id"
+organizationName: "name"
+organizationStreetAddress: "address.line"
+organizationCity: "address.city"
+organizationState: "address.state"
+organizationZip: "referenceRange.postalCode"
+organizationPhoneNumber: "telecom[0].value"
```
[the remark this answers is not in the document]
Comment (nickbristow) [reply containers/ecr-viewer/src/app/api/fhirPath.yml]: no, i removed

File: containers/ecr-viewer/src/app/view-data/components/LabInfo.tsx (hunk 1 above)
Comment (nickbristow) [reply containers/ecr-viewer/src/app/view-data/components/LabInfo.tsx]: 👍

File: containers/ecr-viewer/src/app/view-data/components/tests/LabInfo.test.tsx (hunk 2 above)
Comment (nickbristow) [reply containers/ecr-viewer/src/app/view-data/components/tests/LabInfo.test.tsx:28]: was an issue with getting the click to work with multiple. But yes we can use user.

File: containers/ecr-viewer/src/app/labs/utils.tsx [hunk 4]
```diff
@@ -81,7 +90,10 @@ export const getLabJsonObject = (
   const labsJson = formatTablesToJSON(labsString);
 
   // Get specified lab report (by reference value)
-  return labsJson.filter((obj) => obj.resultId.includes(observationRefVal))[0];
+  // WIP: Issue with the null check
```
[the remark this answers is not in the document]
Comment (KennethSkylight) [reply containers/ecr-viewer/src/app/labs/utils.tsx:93]: Nope I will remove it

**Comment by KennethSkylight** [comment]
Here's the sidenav and a 30 character limit
image

Two orgs: 
image

**Comment by BobanL** [comment]
> ## Two orgs:
> image
> 

@KennethSkylight  - is this real data that we saw? Those tables look not correct with just a title of phOfUrineStrip and the number 5 inside. I don't think any of your changes caused this though - if it is real data it may be worth creating another ticket to address it.

**Comment by KennethSkylight** [comment]
@BobanL it's not real data, I couldn't find an example with two orgs within diagnostic reports in our sample so I created some dummy data

**Review by BobanL — APPROVED** [review]
👍

**Pull request merged**